### PR TITLE
'description-missing' Suggestions, bulk auto fixes, first round. (ML)

### PR DESCRIPTION
**Pull request by MightyPen**
## Summary

'description-missing' Suggestions, bulk auto fixes, first round. (ML)
These build Suggestions were activated 2020/Aug/20, and are scheduled to escalate to Warnings in late October 2020.

I discussed this demo PR with Mark LeBlanc. **I will not merge this PR without Mark LeBlanc's prior OK.**

_(Not a GitHub Issue.)_ Fixes #Issue_Number (if available)

GeneMi  (= MightyPen)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=20322) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=20322) before we can accept your contribution.<br/><br/>:x: MightyPen [sign now](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=20322)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docs?pullRequest=20322) it.</sub>

**Comment by MightyPen** [comment]
@mcleblanc @BillWagner 
Hello Mark and Bill, This PR #20322 can be Closed (and thereafter ignored)  -or- Merged,  whichever you prefer.

If you choose to have this PR Closed, your team of FTE writers might thereby be taking on the duty/task of manually fixing the 11,000 _description-missing_ Suggestion messages in repo 'dotnet/docs'.  It is unclear whether there will ever be sufficient vendor resources for them to manually fix all these  (note that other repositories also have thousands of similar Suggestions that also must be fixed).

If you choose to Merge this PR #20322, I would next run the .Exe again, for another 980 .md files, and repeat the whole run-PR process several times, until all 'description-missing' Suggestions are fixed.

Please let me know how you would like to proceed. I have been told to keep MartinEk and his LT informed how each major repo is progressing on their many Suggestion (and Warning) messages.

It might be possible for me to allocate vendor resources to fix the fewer number of other new Suggestion messages in this repo.

---

_Suggestion &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Count_
description-missing &nbsp;&nbsp;&nbsp;&nbsp; 10043
column-header-missing &nbsp;&nbsp;&nbsp;&nbsp; 661
duplicate-headings &nbsp;&nbsp;&nbsp;&nbsp;&nbsp; 519
image-alt-text-duplicated &nbsp;&nbsp;&nbsp;&nbsp; 130
duplicate-h1s &nbsp;&nbsp;&nbsp;&nbsp; 122
duplicate-titles &nbsp;&nbsp;&nbsp;&nbsp; 42
duplicate-descriptions &nbsp;&nbsp;&nbsp;&nbsp; 17

---

Thanks.  GeneMi (= MightyPen)
. .

File: docs/csharp/language-reference/keywords/add.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ---
+description: "add - C# Reference"
```
Comment (Youssef1313) [inline docs/csharp/language-reference/keywords/add.md:2]:
```suggestion
description: Learn how to create custom event accessors using add keyword in C#
```

**Review by Youssef1313 — CHANGES_REQUESTED** [review]

**Comment by MightyPen** [comment]
@Youssef1313 @mcleblanc 
Youssef, Your recommended changes to the new 'description:' values in this PR #20322 may well be better than the bulk auto values this PR has to offer. In particular, the keywords "custom event accessors" seems plenty desirable to have (as one of your examples).

But this repo has 11,000 'description-missing' Suggestion messages to fix,  plus over 1,000 other messages to fix,  all in the next two months.  We must balance the ideal vs the practical.  No final decision yet.

Thanks.

**Comment by Youssef1313** [comment]
@MightyPen I just picked small amount of files to make this balance. Obviously it' too hard to manually write this large amount of descriptions.

File: docs/csharp/language-reference/builtin-types/numeric-conversions.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ---
+description: "Built-in numeric conversions - C# reference"
```
[the remark this answers is not in the document]
Comment (MightyPen) [reply docs/csharp/language-reference/builtin-types/numeric-conversions.md:2]: @Youssef1313,
I would love to have this kind of superior manually devised/written 'description:' for all 10,637 articles that presently lack the required metadata line for 'description:'. But Mark and I agree that the huge amount of labor to manually devise/write all these would be prohibitive, given our many other pressing priorities.

I will commit your excellent suggested adjustment of this PR #20322, and I will commit the handful of your other suggested adjustments too. Then I will merge this PR.

Thanks a lot.  G

File: docs/csharp/language-reference/compiler-messages/cs0007.md [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ---
+description: "Compiler Error CS0007"
```
[the remark this answers is not in the document]
Comment (MightyPen) [reply docs/csharp/language-reference/compiler-messages/cs0007.md:2]: The situation is explained in another Comment.  Thanks.

**Review by BillWagner — APPROVED** [review]
This is great @MightyPen 

Also, thanks for the extra suggestions @Youssef1313  Those are an improvement.  As mentioned, we don't think it will be worth the investment to hand edit every article. When you do have suggestions, making them is great, but I wouldn't spend too much time on it.

I'll :shipit: now, and keep :eyes: out for upcoming PRs.

Thanks again.

**Comment by MightyPen** [comment]
@mcleblanc @BillWagner Hi, Just fyi, I have been asked to slooow down the bulk auto fixes of 'description-missing' until after the IGNITE conference is completed, on 2020/Sept/24.  The Localization system would become clogged at a time when any IGNITE changes need to be processed promptly.

**Pull request merged**